src/server/pfs/server/compaction.go

```diff
@@ -218,7 +216,9 @@ func (c *compactor) Validate(ctx context.Context, taskDoer task.Doer, id fileset
 	var size int64
 	for i, result := range results {
 		if errStr == "" && i != 0 {
-			errStr = checkIndex(results[i-1].Last, result.First)
+			if indxErr := fileset.CheckIndex(results[i-1].Last, result.First); indxErr != nil {
```

Comment:
nit, I generally think it is better to just use `err` when working within a smaller scope (i.e. within the if statement), and only use specially named errors when necessary and in a larger scope.

Reply:
I was worried `err` would cause some confusion with the outer scope err, though they don't conflict. I'll fix this if that is not a problem.